src/ossl/aes.rs

```diff
@@ -276,8 +276,7 @@ impl AesOperation {
                 let l = 15 - params.ulNonceLen;
                 if params.ulDataLen == 0
                     || params.ulDataLen > (1 << (8 * l))
-                    || (params.ulDataLen + params.ulMACLen)
-                        > CK_ULONG::try_from(u64::MAX)?
+                    || params.ulDataLen > CK_ULONG::MAX - params.ulMACLen
```

Comment:
I assume the operator precedence in rust is sensible (unlike in some other insane languagues), but I would prefer using braces around the expression to be explicit here:
```suggestion
                    || params.ulDataLen > (CK_ULONG::MAX - params.ulMACLen)
```
same in the next chunk

Reply:
the compiler will complain of unneeded parenthesis if I do that, remember I do not establish the rule, `make fix-format` and `rustc` reigns here.